### DISCUSSION TITLE
Always dispaly @odata.id, or do not display @odata.editLink

### DISCUSSION THREAD

**Discussion by alifemove** [OData/AspNetCoreOData#1011] [General] [answered]
Is it possible to either always display `@odata.id` without having to specify `accept: applicaiton/json;odata.metadata=full` or can I at least NOT display `@odata.editLink` when using that header just so users don't get confused as to why that is there.

**Comment by mikepizzo** [comment, marked as answer]
Note that there is [discussion](https://issues.oasis-open.org/browse/ODATA-1529?jql=project%20%3D%20ODATA%20AND%20text%20~%20%22%40id%22) in the OASIS OData Technical Committee on providing a way through the protocol to explicitly request the @odata.id, separate from the other metadata, for example, $select=$ref, or odata.metadata=id.  The issue has been that some services don't return @odata.id, and instead always return key properties.

**Reply by alifemove** [reply]
Sounds like that means it should just be an optional feature lol. At this point we have it in the code to just add the odata.id to every record. Just thought there might be some setting or something that I could use to get it to do it automatically.

Thanks

**Comment by habbes** [comment]
@alifemove you may be able to achieve what you're looking for by extending the `ODataResourceSerializer` with a custom implemntation. You can override the `virtual CreateResource()` method and modify the `ODataResource` object returned from the base class's method. You can either set the `ODataResource.Id` property or remove the `ODataResource.EditLink` property. 

If you try to retrieve the `ODataResource.Id` before you have set it, it will try to compute it automatically if there's sufficient information to compute it. If that succeeds you can set the id back to the property, e.g.:

```c#
class ODataIdCustomODataResourceSerializer : ODataResourceSerializer
{
      public override ODataResource CreateResource(...)
      {
             var resource = base.CreateResource(...);
             var id = resource.Id;
             resource.Id = id;
             return resource;
      }
}
```

The following blog post shows an end-to-end example of creating and configuring a custom serializer: https://devblogs.microsoft.com/odata/build-formatter-extensions-in-asp-net-core-odata-8-and-hooks-in-odataconnectedservice/
